pipeline/test/services/service-cluster/testOpensearch.sh

```diff
@@ -0,0 +1,426 @@
+            users=$(echo "$rolesmapping" | jq -r ".$configured_mapping_name.users[]")
+            if [[ "$users" != "$configured_users" ]]; then
+                no_error=false
+                debug_msg+="[ERROR] Missing users < ${configured_users//$'\n'/ }> != < $users >, for role mapping: $configured_mapping_name \n"
+            fi
+        else
+            no_error=false
+            debug_msg+="[ERROR] Missing role mapping_name ${roleMapping} \n"
+        fi
+    done
+
+    if $no_error; then
+        echo "success ✔"
+        echo "[DEBUG] User roles are configured correctly"
+    else
+        echo "failure ❌"
+        echo -ne "$debug_msg"
+    fi
+}
+
+check_opensearch_ism() {
+    echo -ne "Checking opensearch Index State Management ... "
+    no_error=true
+    debug_msg=""
+
+    default_policies_enabled=$(yq4 -e '.opensearch.ism.defaultPolicies' "${config['config_file_sc']}")
+    if [[ $default_policies_enabled == "true" ]]; then
+        default_policies=("kubernetes" "kubeaudit" "authlog" "other")
+        for policy in "${default_policies[@]}"; do
+            res=$(curl -w "%{http_code}" -o /dev/null -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_plugins/_ism/policies/${policy}")
+            if [[ $res != "200" ]]; then
+                no_error=false
+                debug_msg+="[ERROR] Missing default policy : ${policy}\n"
+            fi
+        done
+    fi
+
+    # additional_policies=$(yq4 -e '.opensearch.ism.additionalPolicies | keys' "${config['config_file_sc']}")
+    readarray -t additional_policies < <(yq4 e -o=j -I=0 '.opensearch.ism.additionalPolicies | keys' "${config['config_file_sc']}")
+
+    additional_policies=("${additional_policies//,/ }")
+    additional_policies=("${additional_policies##[}")
+    additional_policies=("${additional_policies%]}")
+    # additional_policies=($additional_policies)
+    read -ra additional_policies <<<"${additional_policies[@]}"
+
+    for policy in "${additional_policies[@]}"; do
+        policy_name=$(echo "$policy" | awk -F'.' '{print $1}' | tr '"' ' ')
+        res=$(curl -w "%{http_code}" -o /dev/null -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_plugins/_ism/policies/${policy_name}")
+        if [[ $res != "200" ]]; then
+            no_error=false
+            debug_msg+="[ERROR] Missing additional policy : ${policy_name}\n"
+        fi
+    done
+
+    indices_data=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_cat/indices" | awk '{print $3}' | tr '\n' ' ')
+    IFS=' ' read -ra indices <<<"$indices_data"
+
+    for index in "${indices[@]}"; do
+        index_data=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/$index")
+        index_alias=$(echo "$index_data" | jq -r ".\"$index\".settings.index.plugins.index_state_management.rollover_alias")
+        is_write_index=$(echo "$index_data" | jq -r ".\"$index\".aliases.$index_alias.is_write_index")
```

Comment:
Then you can simplify this. And if you want to further optimise I'd suggest using CAT to only fetch the milliseconds creation date using `_cat/indices/<index>?h=creation.date`.